python/src/opendp/extras/polars/__init__.py

```diff
@@ -996,40 +1089,136 @@ class Margin:
+
+    This is used to resolve issues raised in the paper
+    `Widespread Underestimation of Sensitivity in Differentially Private Libraries and How to Fix It <https://arxiv.org/pdf/2207.10635.pdf>`_.
+    """
+
+    max_groups: int | None = None
+    """An upper bound on the number of distinct groups."""
+
+    invariant: Literal["keys"] | Literal["lengths"] | None = None
+    """Identifies properties of grouped data that are considered invariant.
     
     * ``"keys"`` designates that keys are not protected
     * ``"lengths"`` designates that both keys and partition lengths are not protected
+
+    By the analysis of invariants conducted in
+    `Formal Privacy Guarantees with Invariant Statistics <https://arxiv.org/abs/2410.17468>`_,
+    when invariants are set, the effective privacy guarantees of the library are weaker than advertised.
     """
 
-    max_partition_length: int | None = None
-    """An upper bound on the number of records in any one partition.
+    @property
+    @deprecated(reason="Use max_length instead.")
+    def max_partition_length(self):
+        return self.max_length  # pragma: no cover
 
-    If you don't know how many records are in the data, you can specify a very loose upper bound,
-    for example, the size of the total population you are sampling from.
+    @max_partition_length.setter
+    @deprecated(reason="Use max_length instead.")
+    def max_partition_length(self, value):
+        self.max_length = value  # pragma: no cover
 
-    This is used to resolve issues raised in the paper
-    `Widespread Underestimation of Sensitivity in Differentially Private Libraries and How to Fix It <https://arxiv.org/pdf/2207.10635.pdf>`_.
+    @property
+    @deprecated(
+        reason="Use max_groups instead. This was renamed to be consistent with Polars terminology."
+    )
+    def max_num_partitions(self):
+        return self.max_groups  # pragma: no cover
+
+    @max_num_partitions.setter
+    @deprecated(
+        reason="Use max_groups instead. This was renamed to be consistent with Polars terminology."
+    )
+    def max_num_partitions(self, value):
+        self.max_groups = value  # pragma: no cover
+
+    @property
+    @deprecated(
+        reason='Use invariant instead. This was renamed because invariants are not "public information". Invariants are "unprotected information".'
+    )
+    def public_info(self):
+        return self.invariant  # pragma: no cover
+
+    @public_info.setter
+    @deprecated(
+        reason='Use invariant instead. This was renamed because invariants are not "public information". Invariants are "unprotected information".'
+    )
+    def public_info(self, value):
+        self.invariant = value  # pragma: no cover
+
+    @property
+    def max_partition_contributions(self):
+        raise NotImplementedError(
+            "max_partition_contributions has been moved. Use `dp.unit_of(contributions=[dp.polars.Bound(per_group=...)])` instead."
+        )  # pragma: no cover
```

Comment:
I like the messaging here. I wouldn't add any new trivial tests just to cover this, but presumably there are old tests that would have hit this.

One case I'd worry about would be a typo in the the name of this function, and it wouldn't be providing the user assistance we were expecting.